_rules/aria-state-or-property-permitted-5c01ea.md

```diff
@@ -49,81 +50,65 @@ _There are no major accessibility support issues known for this rule._
 
 #### Passed Example 1
 
-`aria-pressed` state is supported for role `button`.
+The `aria-pressed` [state][] is [supported][] for the [semantic role][] `button` which is the [explicit role][] of this `div` element.
 
 ```html
 <div role="button" aria-pressed="false">My button</div>
 ```
 
 #### Passed Example 2
 
-`aria-pressed` state is supported for `role=button` that is the [implicit role](#implicit-role) for `button` element.
+The `aria-pressed` [state][] is [supported][] for the [semantic role][] `button` which is the [implicit role][] for `button` elements.
 
 ```html
 <button aria-pressed="false">My button</button>
 ```
 
 #### Passed Example 3
 
-Global state that is supported by all base markup elements.
+The `aria-busy` [state][] is a [global][] [state][] that is [supported][] by all elements, even without any [semantic role][].
 
 ```html
 <div aria-busy="true">My busy button</div>
 ```
 
 #### Passed Example 4
 
-`aria-label` state is inherited for role `button`
+The `aria-label` [state][] is [inherited][] for the [semantic role][] `button`.
 
 ```html
 <div role="button" aria-label="OK"></div>
 ```
 
 #### Passed Example 5
 
-`aria-checked` state is required for role `aria-checkbox`
+The `aria-checked` [state][] is [required][] for the [semantic role][] `aria-checkbox`.
 
 ```html
 <div role="checkbox" aria-checked="false">My checkbox</div>
 ```
 
 #### Passed Example 6
 
-`aria-controls` property is supported for role `combobox`
+The `aria-controls` [property][] is [supported][] for the [semantic role][] `combobox`.
 
 ```html
 <div role="combobox" aria-controls="id1">My combobox</div>
 ```
 
-#### Passed Example 7
-
-WAI-ARIA states and properties with empty value are also applicable to this rule
-
-```html
-<div role="combobox" aria-controls>My combobox</div>
-```
-
-#### Passed Example 8
-
-WAI-ARIA states and properties with empty value, specified as an empty string, are also applicable to this rule
-
-```html
-<div role="combobox" aria-controls="">My combobox</div>
-```
-
 ### Failed
 
 #### Failed Example 1
 
-`aria-sort` property is neither inherited, supported, nor required for role `button`.
+The `aria-sort` [property][] is neither [inherited][], [supported][], nor [required][] for the [semantic role][] `button`.
 
 ```html
 <div role="button" aria-sort="">Sort by year</div>
 ```
 
 #### Failed Example 2
 
-`aria-sort` attribute is neither inherited, supported, nor required for `role=button` that is the implicit role for `button` element.
+The `aria-sort` [property][] is neither [inherited][], [supported][], nor [required][] for the [semantic role][] `button` which is the [implicit role][] for the `button` element.
 
 ```html
 <button aria-sort="">Sort by year</button>
```

Comment:
I'm considering removing that example as it does not add much compared to Failed Example 1, especially given that the implicit/explicit role thingie has already been examplified in Passed Examples 1 and 2.

Reply:
Perhaps remove Failed example 1 instead. And swap passed examples 1 and 2. Prioritise/encourage semantic HTML element with implicit role over non-semantic element with role defined.